lib/matchers/indexes/have_index_for.rb

```diff
@@ -0,0 +1,18 @@
+module Mongoid
+  module Matchers
+    module Base
```

Comment:
Modules are namespaces, so `Base` isn't a great choice for a module because it doesn't make great sense to organize all base classes together. If all matchers have a base class then just make this class `Mongoid::Matchers::Base`. If all `HaveIndexFor` have a base, you can call it Mongoid::Matchers::HaveIndexForBase` or even `Mongoid::Matchers::HaveIndexFor::Base` (just declare `HaveIndexFor` as a `Class` for the last case).